server/data/redisClient.test.ts

```diff
@@ -0,0 +1,224 @@
+
+      const result = await redisService.getAllDependencies()
+
+      expect(mockRedisClient.json.get).toHaveBeenCalledWith(
+        expect.objectContaining({ isolated: true }),
+        'dependency:info',
+      )
+      expect(result).toBeDefined()
+      // Dependencies class constructor is called with the mock data
+    })
+
+    it('should handle dependency retrieval errors', async () => {
+      const error = new Error('Failed to fetch dependencies')
+      mockRedisClient.json.get.mockRejectedValue(error)
+
+      await expect(redisService.getAllDependencies()).rejects.toThrow('Failed to fetch dependencies')
+
+      expect(mockLogger.error).toHaveBeenCalledWith(
+        'Failed to get dependency info: Failed to fetch dependencies',
+        error,
+      )
+    })
+  })
+
+  describe('handleError', () => {
+    it('should handle ClientClosedError specifically', async () => {
+      const clientClosedError = new ClientClosedError()
+
+      await redisService.handleError(clientClosedError, 'Test operation failed')
+
+      expect(mockLogger.error).toHaveBeenCalledWith('The client is closed ...RECONNECTING')
+      expect(mockLogger.error).toHaveBeenCalledWith('Test operation failed: The client is closed', clientClosedError)
+    })
+
+    it('should handle generic errors', async () => {
+      const genericError = new Error('Generic Redis error')
+
+      await redisService.handleError(genericError, 'Generic operation failed')
+
+      expect(mockLogger.error).toHaveBeenCalledWith('Generic operation failed: Generic Redis error', genericError)
+    })
+  })
+
+  describe('Real-world usage scenarios', () => {
+    it('should handle version data parsing correctly', async () => {
+      const realVersionData = {
+        'component:hmpps-auth:dev': { v: '2023-12-01.456.def5678', dateAdded: '2023-12-01T10:30:00Z' },
+        'component:hmpps-auth:preprod': { v: '2023-12-01.456.def5678', dateAdded: '2023-12-01T10:30:00Z' },
+        'component:hmpps-auth:prod': { v: '2023-11-28.450.abc1234', dateAdded: '2023-11-28T14:22:00Z' },
+      }
+
+      mockRedisClient.json.get.mockResolvedValue(realVersionData)
+
+      const result = await redisService.readLatest('component:latest')
+
+      expect(result).toEqual({
+        'hmpps-auth:dev': { v: '2023-12-01.456.def5678', dateAdded: '2023-12-01T10:30:00Z' },
+        'hmpps-auth:preprod': { v: '2023-12-01.456.def5678', dateAdded: '2023-12-01T10:30:00Z' },
+        'hmpps-auth:prod': { v: '2023-11-28.450.abc1234', dateAdded: '2023-11-28T14:22:00Z' },
+      })
+    })
+
+    it('should handle multiple stream reads in sequence', async () => {
+      const streamData1 = [{ name: 'stream1', messages: [{ id: '1-0', message: { data: 'first' } }] }]
+      const streamData2 = [{ name: 'stream2', messages: [{ id: '2-0', message: { data: 'second' } }] }]
+
+      mockRedisClient.xRead.mockResolvedValueOnce(streamData1).mockResolvedValueOnce(streamData2)
```

Comment:
I think this test is erring on testing jest functionality.. there's nothing in the unit under test that would behave differently depending on whether you called it once or twice is there?